Client/src/Components/Graph/GraphView.tsx

```diff
@@ -1,235 +1,48 @@
-    const variableNames = []
-    selectedDatasets.forEach(dataset => {
-      dataset.data.variables.forEach(variable => {
-        variableNames.push(variable.name)
-      })
-    })
-    const uniqueVariables = Array.from(new Set(variableNames))
-    setVariables(uniqueVariables)
-  }
-
-  const isInStateAlready = (dataset: IDatasetModel) => {
-    return completeDatasets.findIndex(existingDataset => existingDataset.id === dataset.id) != -1
-  }
-
-  const changeAxes = (newAxes: IVariableAndUnitModel) => {
-    setAxes(newAxes)
-  }
-
-  const calculateExtremeBoundaries = (datasets: IDatasetModel[]) => {
-    let minX = 9000, maxX = 0, minY = 9000, maxY = 0
-
-    const datalist: any[] = []
-    //todo refactor graph else this breaks
-    // datasets.forEach(dataset => datalist.push(dataset.points))
-    datalist.forEach(dataset => {
-      dataset.forEach(point => {
-        if (point.x > maxX) {
-          maxX = point.x
-        }
-        if (point.x < minX) {
-          minX = point.x
-        }
-        if (point.y > maxY) {
-          maxY = point.y
-        }
-        if (point.y < minY) {
-          minY = point.y
-        }
-      })
-    })
-    const extremeBoundaries: IDataPointExtremes = { minX: minX, maxX: maxX, minY: minY, maxY: maxY }
-    setDatasetBoundaries(extremeBoundaries)
-  }
-
-
   return (
     <>
-      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"></link>
-      <h2>Graphs</h2>
-      <Modal
-        open={openModal}
-        onClose={handleClose}
-        className={classStyles().modalsearch}
-      >
-        <Box m={5} mb={10} width='100%'>
-          <Grid item xs={12}>
-            <Paper elevation={3}>
-              <Box m={0} p={2}>
-                <CancelIcon color="primary" className={classStyles().closeButton} width="2%" onClick={handleClose} />
-                <Grid container>
-                  <SearchView
-                    handleDatasetsSelected={handleDatasetsSelected}
-                  />
-                </Grid>
-              </Box>
-            </Paper>
-          </Grid>
-        </Box>
-      </Modal>
-      <Box ml={8} mr={8} maxWidth>
-        <Grid container xs={12}>
-          <Grid container xs={9}>
-            <Box m={2} maxWidth mb={20} p={1} pb={10} overflow="hidden">
-              <Paper elevation={3}>
-                <Graph
-                  outerHeight={450}
-                  outerWidth={800}
-                  datasets={completeDatasets.map((dataset, i) => toGraphDataset(dataset, defaultColors[i]))}
-                  extremeBoundaries={datasetBoundaries}
-                />
-              </Paper>
-            </Box>
+      <Box ml={8}>
+        <Grid container spacing={3}>
+          <Grid item container sm={7} >
+            {graphDatasets && graphDatasets[0] ?
+              <Graph
+                datasets={graphDatasets}
+                axes={graphState.axes}
+              /> : null}

```

Comment:
Not a big fan of not displaying a graph if its null... rather have an empty graph. I assume the user will be able to load an empty graph without too much complaint on how much power it takes to load it

Reply:
resolved